simulation/traffic_simulator/src/entity/entity_base.cpp

```diff
@@ -217,6 +217,37 @@ auto EntityBase::getLaneletPose(double matching_distance) const
   }
 }
 
+auto EntityBase::refillEntityStatusWithLaneletData(
+  traffic_simulator_msgs::msg::EntityStatus & status) const -> void
+{
+  const auto unique_route_lanelets = traffic_simulator::helper::getUniqueValues(getRouteLanelets());
+
+  std::optional<traffic_simulator_msgs::msg::LaneletPose> lanelet_pose;
+
+  if (unique_route_lanelets.empty()) {
+    lanelet_pose =
+      hdmap_utils_ptr_->toLaneletPose(status.pose, getStatus().bounding_box, false, 1.0);
```

Comment:
```
hdmap_utils_ptr_->toLaneletPose(status.pose, getStatus().bounding_box, false, 1.0);
```
3 rd argument of this function means include crosswalk lanelet or not.
For pedestrian entity, it should be true.

Reply:
The function `EntityBase::fillLaneletPose()` is supposed to be used only by the EGO entity. I've moved the function from `EntityBase` class to `EntityManager` class -> 70b2f45bc2993f718afa77bd367fd7d9ec2bd10d

Reply:
If this function is only for ego entity, it should be moved to Ego entity class, not a entity manager class.

Reply:
I've moved `fillLaneletPose()` to `EntityBase` as a virtual method, and implemented the `fillLaneletPose()` for `EgoEntity`  -> 7f12513d1c0b8c32ef278966ac991fdf275f79a7

Reply:
Thanks!